File: robot/shared/Shape/Circle.cpp
#include "Circle.h"

Circle::Circle(const Vector2D & _center, float _radius,
               int fillColor)
  : Shape(Shape::Circle, fillColor),
    center(_center),
    radius(_radius) {
}

Circle::Circle(const Vector2D & _center, float _radius,
               int strokeColor, float strokeThickness)
  : Shape(Shape::Circle, strokeColor, strokeThickness),
    center(_center),
    radius(_radius) {
}

Circle::Circle(const Vector2D & _center, float _radius,
               int fillColor,
               int strokeColor, float strokeThickness)
  : Shape(Shape::Circle, fillColor, strokeColor, strokeThickness),
    center(_center),
    radius(_radius) {
}

Circle::~Circle() {
}

const Vector2D & Circle::getCenter() const {
  return center;
}

float Circle::getRadius() const {
  return radius;
}

File: tools/Visualizer/src/shape/Circle.java
package shape;

public class Circle extends Shape {
  private float centerX, centerY;
  private float radius;

  public Circle(float centerX, float centerY,
                float radius,
                int fillColor,
                int strokeColor, float strokeThickness) {
    super(fillColor, strokeColor, strokeThickness);
    this.centerX = centerX;
    this.centerY = centerY;
    this.radius = radius;
  }

  public Circle(float centerX, float centerY,
                float radius,
                int strokeColor, float strokeThickness) {
    super(strokeColor, strokeThickness);
    this.centerX = centerX;
    this.centerY = centerY;
    this.radius = radius;
  }

  public Circle(float centerX, float centerY,
                float radius,
                int fillColor) {
    super(fillColor);
    this.centerX = centerX;
    this.centerY = centerY;
    this.radius = radius;
  }

  public float getCenterX() {
    return centerX;
  }

  public float getCenterY() {
    return centerY;
  }

  public float getRadius() {
    return radius;
  }

}
